### ISSUE TITLE
Flutter CLI cannot code sign builds and therefore cannot install on physical development device 

### ISSUE THREAD

**Issue by mattbodley** [flutter/flutter#137986]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

flutter run --flavor qa -t lib/main_qa.dart

The "It's also possible that a previously installed app with the same Bundle 
Identifier was signed with a different certificate." makes the most sense to reproduce this. As recently the certificate had expired and was refreshed. This should be done automatically e.g. in Xcode it is automatically done but there seems to be a disconnect with flutter run as flutter cannot detect or or at least cannot detect the right one. 

### Expected results

It should run!

### Actual results

Automatically signing iOS for device deployment using specified development team in Xcode project: xxxxxxxxx
Running Xcode build...
Xcode build done.                                           178.1s
Installing and launching...
════════════════════════════════════════════════════════════════════════════════
No Provisioning Profile was found for your project's Bundle Identifier or your 
device. You can create a new Provisioning Profile for your project in Xcode for 
your team by:
  1- Open the Flutter project's Xcode target with
       open ios/Runner.xcworkspace
  2- Select the 'Runner' project in the navigator then the 'Runner' target
     in the project settings
  3- Make sure a 'Development Team' is selected under Signing & Capabilities > Team. 
     You may need to:
         - Log in with your Apple ID in Xcode first
         - Ensure you have a valid unique Bundle ID
         - Register your device with your Apple Developer Account
         - Let Xcode automatically provision a profile for your app
  4- Build or run your project again

It's also possible that a previously installed app with the same Bundle 
Identifier was signed with a different certificate.

For more information, please visit:
  https://flutter.dev/docs/get-started/install/macos#deploy-to-ios-devices

Or run on an iOS simulator without code signing
════════════════════════════════════════════════════════════════════════════════

### Code sample

<details><summary>Code sample</summary>

```dart
[Paste your code here]
```

</details>

### Screenshots or Video

<details>
<summary>Screenshots / Video demonstration</summary>

**Failed run via CLI using Flutter**

![Screen Shot 2023-11-07 at 8 38 11 PM](https://github.com/flutter/flutter/assets/3054509/40d878ac-6edb-4c97-9a08-46087cd1174a)

**Xcode is Setup as specified in Flutter CLI output**

![Screen Shot 2023-11-07 at 8 40 09 PM](https://github.com/flutter/flutter/assets/3054509/65729401-cacc-418b-b3b1-7e7cac534a6d)
![Screen Shot 2023-11-07 at 8 40 16 PM](https://github.com/flutter/flutter/assets/3054509/2e4eb2bc-f69d-47a5-a9c9-af417a958035)

**Xcode build works properly**

![Screen Shot 2023-11-07 at 8 47 09 PM](https://github.com/flutter/flutter/assets/3054509/190c923b-88dd-4395-a08c-40141b6beac3)

</details>

### Logs

<details><summary>Logs</summary>

```console
      No Provisioning Profile was found for your project's Bundle Identifier or your 
                    device. You can create a new Provisioning Profile for your project in Xcode for 
                    your team by:
                      1- Open the Flutter project's Xcode target with
                           open ios/Runner.xcworkspace
                      2- Select the 'Runner' project in the navigator then the 'Runner' target
                         in the project settings
                      3- Make sure a 'Development Team' is selected under Signing & Capabilities > Team. 
                         You may need to:
                             - Log in with your Apple ID in Xcode first
                             - Ensure you have a valid unique Bundle ID
                             - Register your device with your Apple Developer Account
                             - Let Xcode automatically provision a profile for your app
                      4- Build or run your project again
                    
                    It's also possible that a previously installed app with the same Bundle 
                    Identifier was signed with a different certificate.
                    
                    For more information, please visit:
                      https://flutter.dev/docs/get-started/install/macos#deploy-to-ios-devices
                    
                    Or run on an iOS simulator without code signing
```

</details>

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
[✓] Flutter (Channel stable, 3.13.9, on macOS 12.6.1 21G217 darwin-x64, locale en-NZ)
[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 14.1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2021.3)
[✓] VS Code (version 1.84.0)
[✓] Connected device (4 available)
[✓] Network resources

```

</details>

**Comment by huycozy** [comment]
Hi @mattbodley 
Which iOS version are you checking on? Please try [this](https://github.com/flutter/flutter/issues/136793#issuecomment-1777517406) solution to see if it helps.

**Comment by mattbodley** [comment]
Thanks @huycozy nice spotting! The version of iOS on my device is 16.2 so Flutter should support it via the CLI will the provisioning profile solution now

**Comment by mattbodley** [comment]
@huycozy I just tried the flutter attach solution mentioned and that seems to work too, although a bit clunky as not as integrated (the debug logs are in Xcode, and the hot reload/restart need to be done in the terminal via R/r) but essentially appears to be the same result as running in debug mode within Android Studio

**Comment by mattbodley** [comment]
@huycozy it seems that the flutter attach will not do hot reload/restart with any changed files which is an issue too

**Comment by mattbodley** [comment]
@huycozy I tried regenerating profiles in Xcode and importing them into onto the device and it just comes up with an error:

![Screen Shot 2023-11-09 at 8 02 24 AM](https://github.com/flutter/flutter/assets/3054509/3706b622-3949-4f99-8314-
c5c38df4d1d2)

Although, it still seems to run fine when I build in Xcode even with the certificates deleted as per 

https://developer.apple.com/help/account/manage-provisioning-profiles/edit-download-or-delete-profiles#:~:text=provisioning%20profile.-,Regenerate%20a%20provisioning%20profile,-If%20you%20enable 

The existing profiles on the device are not expired and are:
iOS Team Provisioning Profile: *
iOS Team Provisioning Profile: co.domain.name.app: 
App name: ios_app_store_1698127211

What does that mean? What do you think are the next steps? These are not expired if I delete these then it won't work at all right? And it comes up with the error above in the screenshot when I tried to install them.

The **Xcode managed signing** is still checked in Xcode and it still builds OK

**Comment by huycozy** [comment]
There may be some causes for that error: device is not connected or selected device is not used in new provisioning profile. Please try checking [this](https://stackoverflow.com/questions/61220705/xcode-11-4-error-failed-to-install-one-or-more-provisioning-profiles-on-the-de) if one of the solutions there helps.

**Comment by matt-12345678** [comment]
I tried that solution and it didn't seem to make a difference, "Automatically manage signing" is turned off but the result is the same, flutter attach works seems to attach but reload/refresh don't work I think it is trying to copy code over to the device but it is being rejected. 

One thing I found is that on [this link](https://docs.flutter.dev/get-started/install/macos?tab=ios16#deploy-to-ios-devices ) 

![Screen Shot 2023-11-09 at 9 10 51 PM](https://github.com/flutter/flutter/assets/3054509/2b3970fc-7477-4107-9589-70fa58d2e1a2)

My device doesn't have this setting and it is iOS 16.2 - why is it like this?

![IMG_5647](https://github.com/flutter/flutter/assets/3054509/b1105cac-73a9-4d51-9e7a-808181d532a3)

**Comment by huycozy** [comment]
>flutter attach works seems to attach but reload/refresh don't work

Does that mean there is nothing happens when performing `hot reload`/`restart`? I would like to confirm again, do you still see the initial error?

>My device doesn't have this setting and it is iOS 16.2 - why is it like this?

This could be a website issue. Please help to file an issue on [website repository](https://github.com/flutter/website/issues) for this. Thanks!

**Comment by matt-12345678** [comment]
yes there is still the issue, I can build in xcode and then when I attach via **flutter attach** it will exit when it attempts to do a reload/restart

**Comment by huycozy** [comment]
Even though this issue has a similar error message to https://github.com/flutter/flutter/issues/136793, the case is not the same (XCode/OS version). Keeping this as a separate issue for further input.

**Comment by mattbodley** [comment]
@huycozy yes interesting issue, I think that there is probably a bug with the flutter code as it is not getting the Xcode profile properly. Probably due to layers of upgrades over time. I have reinstalled everything and it just works! I still have the old install that has the same issue but on the new install it is working fine. It would be interesting to know the cause of the issue, I can replicate it but it just cannot be replicated on other machines which is not good.

**Comment by hellohuanlin** [comment]
cc: @vashworth

**Comment by vashworth** [comment]
> @huycozy yes interesting issue, I think that there is probably a bug with the flutter code as it is not getting the Xcode profile properly. Probably due to layers of upgrades over time. I have reinstalled everything and it just works! I still have the old install that has the same issue but on the new install it is working fine. It would be interesting to know the cause of the issue, I can replicate it but it just cannot be replicated on other machines which is not good.

@mattbodley What do you mean by "old install" and "new install"? Install of the app on the device? Or new install of flutter?
